lib/spack/spack/main.py

```diff
@@ -900,9 +900,8 @@ def add_command_line_scopes(
 
     for i, path in enumerate(command_line_scopes):
         # If an environment is set on the CLI, add its scope in the order it appears there.
-        # Subsequent custom scopes will override it, and it will override prior custom scopes.
         if path is _ENV:
-            add_environment(ConfigScopePriority.CUSTOM)
+            add_environment(ConfigScopePriority.ENVIRONMENT)
```

Comment:
I suspect some of this code can be simplified now that envs only come in at one priority, but that can wait if this is the best way to get the bug fixed for now.